### PR TITLE
Add support for deserialization of owned values

### PR DESCRIPTION
**Pull request by nklhtv**
Based on PR #8 but includes:
- adapt to the new serde api
- fixes for SerHexOpt 
- additional test
- cargo fmt

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ use smallvec::SmallVec;
 use std::iter::FromIterator;
 use std::{error, io, fmt};
 use serde::de::{Visitor};
-use serde::export::{PhantomData};
+use serde::__private::{PhantomData};
```
Comment (thomaseizinger) [inline src/lib.rs:57]: Well, we should really be importing this from std, haha.

File: src/lib.rs [hunk 2]
```diff
@@ -49,12 +49,12 @@ pub mod utils;
 pub use config::*;
 pub use types::{Error, ParseHexError};
 
+use serde::__private::PhantomData;
```
Comment (thomaseizinger) [inline src/lib.rs]: Invasive things like `cargo fmt` are likely better left to individual PRs, just adds unnecessary noise to the diff and thus work to the reviewers.

**Review by thomaseizinger — COMMENTED** [review]
I've integrated the additional fixes into my PR by cherry-picking your commits!

https://github.com/fspmarshall/serde-hex/pull/8

File: src/lib.rs (hunk 1 above)
Comment (nklhtv) [reply src/lib.rs:57]: @thomaseizinger I agree with you, and i reverted the `code reformat` commit

File: src/lib.rs (hunk 2 above)
Comment (nklhtv) [reply src/lib.rs]: @thomaseizinger I agree with you, and i reverted the `code reformat` commit

**Comment by nklhtv** [comment]
Closed in favor of #8 
essential commits ware cherry-picked

**Pull request closed without merging**